parts/k8s/kubernetesagentcustomdata.yml

```diff
@@ -96,11 +96,13 @@ write_files:
     ExecStartPre=/usr/bin/docker pull {{WrapAsVariable "kubernetesHyperkubeSpec"}}
     ExecStartPre=/usr/bin/docker run --rm -v /tmp/kubectldir:/opt/kubectldir {{WrapAsVariable "kubernetesHyperkubeSpec"}} /bin/bash -c "cp /hyperkube /opt/kubectldir/"
 {{if .IsCoreOS}}
+    ExecStartPre=/bin/cp /tmp/kubectldir/hyperkube /opt/kubelet
     ExecStartPre=/bin/mv /tmp/kubectldir/hyperkube /opt/kubectl
-    ExecStart=/bin/chmod a+x /opt/kubectl
+    ExecStart=/bin/chmod a+x /opt/kubelet && /bin/chmod a+x /opt/kubectl
 {{else}}
+    ExecStartPre=/bin/cp /tmp/kubectldir/hyperkube /usr/local/bin/kubelet
     ExecStartPre=/bin/mv /tmp/kubectldir/hyperkube /usr/local/bin/kubectl
-    ExecStart=/bin/chmod a+x /usr/local/bin/kubectl
+    ExecStart=/bin/chmod a+x /usr/local/bin/kubelet && /bin/chmod a+x /usr/local/bin/kubectl
```

Comment:
We can shorten this to `/bin/chmod a+x /usr/local/bin/kubelet /usr/local/bin/kubectl` (the build of `chmod` on ubuntu hosts accepts a list of file targets to apply the mode changes against). Verified:

```
azureuser@k8s-master-18440544-0:/tmp$ ls -la test*
-rw-rw-r-- 1 azureuser azureuser 0 Apr  3 23:08 test1
-rw-rw-r-- 1 azureuser azureuser 0 Apr  3 23:08 test2
azureuser@k8s-master-18440544-0:/tmp$ /bin/chmod a+x test1 test2
azureuser@k8s-master-18440544-0:/tmp$ ls -la test*
-rwxrwxr-x 1 azureuser azureuser 0 Apr  3 23:08 test1
-rwxrwxr-x 1 azureuser azureuser 0 Apr  3 23:08 test2
```

Reply:
Yes good catch, thanks